src/adapter/endpoint.ts

```diff
@@ -1,40 +1,75 @@
 import { ResponseCache } from '../cache/response'
 import { AdapterConfig } from '../config'
-import { MetaTransport, Transport } from '../transports'
-import { makeLogger } from '../util'
+import { Transport } from '../transports'
+import { AdapterRequest, AdapterRequestData, makeLogger } from '../util'
 import { SpecificInputParameters } from '../validation'
+import { AdapterError } from '../validation/error'
 import { InputValidator } from '../validation/input-validator'
 import {
   AdapterDependencies,
+  AdapterEndpointInterface,
   AdapterEndpointParams,
   CustomInputValidator,
   EndpointGenerics,
   EndpointRateLimitingConfig,
+  Overrides,
+  RequestTransform,
 } from './types'
 
 const logger = makeLogger('AdapterEndpoint')
+const DEFAULT_TRANSPORT_NAME = 'default_single_transport'
+
 /**
  * Main class to represent an endpoint within an External Adapter
  */
-export class AdapterEndpoint<T extends EndpointGenerics> implements AdapterEndpointParams<T> {
+export class AdapterEndpoint<T extends EndpointGenerics> implements AdapterEndpointInterface<T> {
   name: string
   aliases?: string[] | undefined
-  transport: Transport<T> | MetaTransport<T>
+  transports: Record<string, Transport<T>>
   inputParameters: SpecificInputParameters<T['Request']['Params']>
   rateLimiting?: EndpointRateLimitingConfig | undefined
   validator: InputValidator
   cacheKeyGenerator?: (data: Record<string, unknown>) => string
   customInputValidation?: CustomInputValidator<T>
```

Comment:
Now that the transport field isn't in the input params, `customInputValidation` can't validate which transport the request is going to since its parameters are input params and config. Think we may need to update it to have transportName that we can pass in the validator. We need it for adapters that have optional WS transports to be able to throw an error if a request comes in for ws when the api key isn't set.

Reply:
Noticed this when I was trying the new changes on `tradermade-test`

Reply:
Valid point, I've given the custom input validator the entire request, don't really love the patterns that could come of it, but it's going to solve many problems in the short term